File: docs/APIs-and-SDKs/SDK-Documentation/getting-started/custom-event-logger/java/customEventLogger.java
    // example implementation
    public class CustomEventLogger implements ContextEventLogger {
        @Override
        public void handleEvent(Context context, ContextEventLogger.EventType event, Object data) {
            switch (event) {
            case Exposure:
                final Exposure exposure = (Exposure)data;
                System.out.printf("exposed to experiment %s", exposure.name);
                break;
            case Goal:
                final GoalAchievement goal = (GoalAchievement)data;
                System.out.printf("goal tracked: %s", goal.name);
                break;
            case Error:
                System.out.printf("error: %s", data);
                break;
            case Publish:
            case Ready:
            case Refresh:
            case Close:
                break;
            }
        }
    }
    // for all contexts, during sdk initialization
    final ABSmartlyConfig sdkConfig = ABSmartlyConfig.create();
    sdkConfig.setContextEventLogger(new CustomEventLogger());

    // OR, alternatively, during a particular context initialization
    final ContextConfig contextConfig = ContextConfig.create();
    contextConfig.setEventLogger(new CustomEventLogger());

File: docs/APIs-and-SDKs/SDK-Documentation/getting-started/custom-event-logger/dart/customEventLogger.dart
class CustomEventLogger implements ContextEventLogger {
  @override
  void handleEvent(Context context, EventType event, dynamic data) {
    switch (event) {
      case EventType.exposure:
        final Exposure exposure = data;
        print("exposed to experiment ${exposure.name}");
        break;
      case EventType.goal:
        final GoalAchievement goal = data;
        print("goal tracked: ${goal.name}");
        break;
      case EventType.error:
        print("error: $data");
        break;
      case EventType.publish:
      case EventType.ready:
      case EventType.refresh:
      case EventType.close:
        break;
    }
  }
}
